library/std/src/path.rs

```diff
@@ -2632,6 +2702,30 @@ impl Path {
         new_path
     }
 
+    /// Creates an owned [`PathBuf`] like `self` but with an extra extension.
+    ///
+    /// See [`PathBuf::add_extension`] for more details.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::path::{Path, PathBuf};
+    ///
+    /// let path = Path::new("foo.rs");
+    /// assert_eq!(path.with_extra_extension("txt"), PathBuf::from("foo.rs.txt"));
+    ///
+    /// let path = Path::new("foo.tar.gz");
+    /// assert_eq!(path.with_extra_extension(""), PathBuf::from("foo.tar.gz"));
+    /// assert_eq!(path.with_extra_extension("xz"), PathBuf::from("foo.tar.gz.xz"));
+    /// assert_eq!(path.with_extra_extension("").with_extra_extension("txt"), PathBuf::from("foo.tar.gz.txt"));
+    /// ```
+    #[stable(feature = "rust1", since = "1.0.0")]
+    pub fn with_extra_extension<S: AsRef<OsStr>>(&self, extension: S) -> PathBuf {
```

Comment:
Some how we can simplify `with_extension` in:

```rust
    pub fn with_extension<S: AsRef<OsStr>>(&self, extension: S) -> PathBuf {
        let mut new_path = self.to_path_buf();
        new_path.set_extension(extension);
        new_path
    }
```

But it's out of the scope of this PR so I just mention it here. Or I omit some performance overhead here?